Application/src/model/Menu.java

```diff
@@ -338,4 +433,159 @@ public void identifyState() {
+        grid.add(pw, 0, 0);
+
+        PasswordField pwBox = new PasswordField(); // Textfield with password protection.
+        grid.add(pwBox, 1, 0);
+        pwBox.setPromptText("Password");
+
+        String ok_password_text = "OK";
+        Button button_ok_password = new Button(ok_password_text); // Button to create a lobby with input password.
+        grid.add(button_ok_password, 1,1);
+
+        final Text pressReturn = new Text(); //
+        grid.add(pressReturn, 1, 2);
+
+        //Handler for pressing the ok button in the newly opened window.
+        button_ok_password.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent e) {
+                final String result = pwBox.getText();
+                if(result.equals("")) pressReturn.setText("Please enter a password");
+                else {
+                    Net.push("{" + "share, " + result + ", create_lobby}");
+                    host = true;
+                    primaryStage.close();
+                }
+            }
+        });
+    }
+
+    /**
+     * Creates a separate window where a user enters the desired lobby to leave.
+     */
+    public void leaveLobbyBox(){
+        String ok_leave_text = "OK";
+        Stage leaveStage = new Stage(); // Creates a new 'stage' where we gather the fields and buttons for the box.
+        leaveStage.setTitle("Input Lobby ID");
+        leaveStage.show();
+
+        GridPane grid = new GridPane(); // Creates a GridPane to easily place the different components.
+        grid.setAlignment(Pos.CENTER);
+        //Sets a "padding" which makes the TexField and related components more centered and not use the entire box.
+        grid.setHgap(10);
+        grid.setVgap(10);
+        grid.setPadding(new Insets(15, 15, 15, 15));
+
+        Scene scene = new Scene(grid, 250, 125);
+        leaveStage.setScene(scene);
+        TextField leaveBox = new TextField();
+        grid.add(leaveBox, 1, 0);
+        leaveBox.setPromptText("Lobby ID");
+        Button button_ok_leave = new Button(ok_leave_text); // Button to leave a lobby with id.
+        grid.add(button_ok_leave, 1,1);
+
+        final Text pressReturn = new Text(); //
+        grid.add(pressReturn, 1, 2);
+
+        //Handler for pressing the ok button in the newly opened window.
+        button_ok_leave.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent e) {
+                final String result = leaveBox.getText();
+                    // Send the result to the server.
+                    Net.push("{" + "share, " + result + ", leave_lobby}");
+                    leaveStage.close();
+                }
+        });
+    }
+
+    /**
+     * Sends a message of the given message type to the server
+     * @param MessageType The message type, either previous_message or next_message
+     */
+    public static void changeMessage(String MessageType){
```

Comment:
ChangeMessage doesn't sound like a class which sends messages to the server. Maybe rename it and move in to the network class too lower menu coupling?